steps/data_embedding_steps/compute_embedding_drift_step/compute_embedding_drift_step.py

```diff
@@ -0,0 +1,154 @@
+
+    Returns:
+        List[float]: the mean of each list of embeddings.
+    """
+    return [mean(embedding) for embedding in embeddings]
+
+
+def calculate_euclidean_distance(
+    reference_embeddings: List[List[float]], current_embeddings: List[List[float]]
+) -> float:
+    """Calculate the Euclidean distance between the mean of reference embeddings and the mean of current embeddings.
+
+    Args:
+        reference_embeddings (List[List[float]]): a list of lists, where each inner list contains floats of reference embeddings
+        current_embeddings (List[List[float]]): a list of lists, where each inner list contains floats of current embeddings
+
+    Raises:
+        ValueError: raise if the len of the reference dataset embedding mean does not equal the length current dataset
+
+    Returns:
+        float: the Euclidean distance between the mean of reference embeddings and the mean of current embeddings
+    """
+    reference_embeddings_mean = calculate_means(reference_embeddings)
+    current_embeddings_mean = calculate_means(current_embeddings)
+
+    reference_lengths = [len(embedding) for embedding in reference_embeddings]
+    current_lengths = [len(embedding) for embedding in current_embeddings]
+
+    if reference_lengths != current_lengths:
+        raise ValueError(
+            "The length of the reference embeddings mean list should equal to the length of the current embeddings mean list"
+        )
+
+    return float(distance.euclidean(reference_embeddings_mean, current_embeddings_mean))
+
+
+def build_embedding_drift_payload(
+    reference_data_version: str, current_data_version: str, distance: float
+) -> Dict[str, Union[str, float, bool]]:
+    """Construct a payload for send the embedding drift data to the metric service via post request.
+
+    Args:
+        reference_data_version (str): the version identifier for the reference data.
+        current_data_version (str): the version identifier for the current data.
+        distance (float): the computed Euclidean distance between the embeddings of the reference and current data.
+
+    Returns:
+        Dict[str, Union[str, float, bool]]: a dictionary containing the 4 items required by the metric service embedding drift relation.
+    """
+    drifted = distance > 0
+
+    formatted_reference_data_version = f"'{reference_data_version}'"
+    formatted_current_data_version = f"'{current_data_version}'"
+
+    return {
+        "reference_dataset": f"{formatted_reference_data_version}",
+        "current_dataset": f"{formatted_current_data_version}",
+        "distance": distance,
+        "drifted": drifted,
+    }
+
+
+@step
+def compute_embedding_drift(
+    collection_name: str, reference_data_version: str, current_data_version: str
+) -> float:
+    """Compute the measure of 'drift' in data embeddings between the current and reference datasets, identified by the given collection name.
```

Comment:
A mention of this step also sends a post request to the monitoring service would be helpful.